### PR TITLE
Fix: `key-spacing.align` doesn't pay attention to non-whitespace before key (fixes #3267)

### LINKED ISSUE #3267: `key-spacing.align` doesn't pay attention to non-whitespace before key
I'll admit my code style is weird, but `key-spacing.align` doesn't actually enforce alignment of values/colons like the documentation states, it enforces the alignment of keys _and_ values/colons.

If this is intended, the docs should be updated (and theoretically the error messages changed).  But I don't think it is:

```js
var x = {
  foo : 10
  , b : 20 // missing space after key
};

var x = {
  foo    : 10, // extra space after key
  /**/ b : 20  // missing space after key
};
```

these pass:

```js
var x = {
  foo : 10
  , b   : 20
};

var x = {
  foo : 10,
  /**/ b   : 20
};
```

what it "wants":

```js
var x = {
    foo : 10
  , b   : 20
};

var x = {
       foo : 10,
  /**/ b   : 20
};
```

eslint 0.24.1, `"key-spacing": { "align" : "colon" }`

### PR DESCRIPTION
**Pull request by BYK**

### CODE REVIEW

File: lib/rules/key-spacing.js [hunk 1]
```diff
@@ -98,6 +98,35 @@ module.exports = function(context) {
...
+            firstToken = context.getTokenBefore(node);
+            startLine = node.loc.start.line;
+            endLine = firstToken ? firstToken.loc.end.line : -1;
+        } while (startLine === endLine);
+
+        return node;
+    }
+
+    function getFirstNodeBeforeColon(node) {
+        var prevNode;
```
Comment (nzakas) [inline lib/rules/key-spacing.js]: Please add JSDoc here

File: lib/rules/key-spacing.js [hunk 2]
```diff
@@ -98,6 +98,35 @@ module.exports = function(context) {
...
+
+        while (node.type !== "Punctuator" || node.value !== ":") {
```
Comment (nzakas) [inline lib/rules/key-spacing.js]: Should be double quotes....not sure why linting didn't pick this up.

Comment (BYK) [inline lib/rules/key-spacing.js]: They are double quotes and the linter did catch that :) I used single quotes first and just run the tests for this rule so missed it. Then CI yelled at me and then I fixed.

File: lib/rules/key-spacing.js (hunk 1 above)
Comment (BYK) [inline lib/rules/key-spacing.js]: Added.

File: lib/rules/key-spacing.js (hunk 2 above)
Comment (nzakas) [inline lib/rules/key-spacing.js]: Ah cool!

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**